portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -240,29 +258,37 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     // Obtain user input from submitted form and timestamp.
     String userId = userService.getCurrentUser().getUserId();
-    String message = request.getParameter("message");
-    String name = request.getParameter("message-sender-name");
+    String placeQueryName = request.getParameter("place-query-name");
+    String commentContent = request.getParameter("comment-content");
+    String name = request.getParameter("comment-sender-name");
     String email = userService.getCurrentUser().getEmail();
     String petPreference = request.getParameter("user-pet-preference");
+    // Get the URL of the image that the user uploaded to Blobstore.
+    String imageUrl = getUploadedFileUrl(request, "comment-image");
     // Refrain from adding to the database if the user input is a potentially XSS attack.
     // Users select petPreference from a set of predefined options, so petPreference is safe.
-    if (!message.equals(Encode.forHtml(message)) || !name.equals(Encode.forHtml(name))) {
+    if (!placeQueryName.equals(Encode.forHtml(placeQueryName)) ||
+        !commentContent.equals(Encode.forHtml(commentContent)) ||
+        !name.equals(Encode.forHtml(name))) {
       this.invalidInputFlags.setIsLatestInputDangerous();
       response.sendRedirect(REDIRECT_URL);
       return;
     }
     long timestamp = System.currentTimeMillis();
 
     // Pack user input into an object.
-    comment = new UserComment(userId, message, name, email, petPreference);
+    comment = new UserComment(userId, placeQueryName, commentContent, name, email, petPreference,
```

Comment:
I think it's time to revisit our "too many parameters" discussion. Consider packaging name, email, petPreference  (maybe even userId) etc into something like UserInfo?

Reply:
Will do! Thanks for the feedback